apps/sms/services/js/messaging/messaging_client.js

```diff
@@ -0,0 +1,110 @@
+/* -*- Mode: js; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- /
+/* vim: set shiftwidth=2 tabstop=2 autoindent cindent expandtab: */
+
+
+/* global bridge, EventDispatcher */
+
+/* export MessagingClient */
+
+'use strict';
+(function(exports) {
+
+const EVENTS = [
+  'message-sending', 'message-failed-to-send', 'message-delivered',
+  'message-read', 'message-sent', 'message-received', 'threads-deleted'
+];
+
+var client = bridge.client('messaging');
+
+var MessagingClient = {
+  relayEvents: function msc_relayEvents() {
+    EVENTS.forEch((event) => {
+      client.on(event, (item) => {
+        this.emit(event, item);
+      });
+    });
+  },
+
+  getMessage: function msc_getMessage(id) {
+    return client.method('getMessage', id);
+  },
+
+  retrieveMMS: function msc_retrieveMMS(id) {
+    return client.method('retrieveMMS', id);
+  },
+
+  sendSMS: function msc_sendSMS(opts) {
+    return client.method('sendSMS', opts);
+  },
+
+  sendMMS: function msc_sendMMS(opts) {
+    return client.method('sendMMS', opts);
+  },
+
+  resendMessage: function msc_resendMessage(opts) {
+    return client.method('resendMessage', opts);
+  },
+
+  deleteMessages: function msc_deleteMessages(id, callback) {
+    return client.method('deleteMessages', id, callback);
+  },
+
+  markThreadRead: function msc_markThreadRead(threadId) {
+    return client.method('markThreadRead', threadId);
+  },
+
+  markMessageRead: function msc_markMessageRead(list) {
+    return client.method('markMessageRead', list);
+  },
+
+  getSegmentInfo: function msc_getSegmentInfo(text) {
+    return client.method('getSegmentInfo', text);
+  },
+
+  findThreadFromNumber: function msc_findThreadFromNumber(number) {
+    return client.method('findThreadFromNumber', number);
+  },
```

Comment:
I'm sure we can try to be more DRY here :) Maybe put all methods in an array ? or use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy ?